keep/providers/auth0_provider/auth0_provider.py

```diff
@@ -18,20 +20,21 @@ class Auth0ProviderAuthConfig:
     Auth0 authentication configuration.
     """
 
-    token: str = dataclasses.field(
-        default=None,
+    domain: HttpsUrl = dataclasses.field(
         metadata={
             "required": True,
-            "description": "Auth0 API Token",
-            "hint": "https://manage.auth0.com/dashboard/us/YOUR_ACCOUNT/apis/management/explorer",
+            "description": "Auth0 Domain",
+            "hint": "https://tenantname.us.auth0.com",
+            "validation": "https_url",
         },
     )
-    domain: str = dataclasses.field(
+
+    token: str = dataclasses.field(
         default=None,
         metadata={
             "required": True,
-            "description": "Auth0 Domain",
-            "hint": "tenantname.us.auth0.com",
+            "description": "Auth0 API Token",
+            "hint": "https://manage.auth0.com/dashboard/us/YOUR_ACCOUNT/apis/management/explorer",
         },
```

Comment:
@coderabbitai I have added unit tests for all custom validation classes.

We do not wish to write end-to-end validation tests for every provider. Kibana, Teams and others were chosen basically because they have some of the custom validation classes in their config fields, other providers have these tested fields as well.